_test/check-exercises.sh

```diff
@@ -1,73 +1,56 @@
 #!/bin/bash
 
+# test for existence and executability of the test-exercise script
+# this depends on that
+if [ ! -f "./bin/test-exercise" ]; then
+   echo "bin/test-exercise does not exist"
+   exit 1
+fi
+if [ ! -x "./bin/test-exercise" ]; then
+   echo "bin/test-exercise does not have its executable bit set"
+   exit 1
+fi
+
 # In DENYWARNINGS mode, do not set -e so that we run all tests.
 # This allows us to see all warnings.
 if [ -z "$DENYWARNINGS" ]; then
     set -e
 fi
 
+# can't benchmark with a stable compiler; to bench, use
+# $ BENCHMARK=1 rustup run nightly _test/check-exercises.sh
 if [ -n "$BENCHMARK" ]; then
     files=exercises/*/benches
+    # cargo bench --release is an invalid command
+    release=""
 else
     files=exercises/*/tests
+    release="--release"
```

Comment:
I proposed this change in https://github.com/exercism/rust/pull/241 and it was rejected, for the reason:

> A reason to not use release mode: it will skip overflow checks

I support this change. Please be aware of the reasons for the previous rejection.

Reply:
I put that in to support testing of the script locally, but I'm only moderately in favor of it. It makes sense to ensure that nothing overflows in the tests, so I'll need to get rid of that.

Reply:
A possible nice change to make would to test in release mode if `.meta/test-in-release-mode` is present, or something along those lines.